module/dkg/controller.go

```diff
@@ -326,3 +380,45 @@ func (c *Controller) phase3() error {
 		}
 	}
 }
+
+// preStartDelay returns a duration to delay prior to starting the DKG process.
+// This prevents synchronization of the DKG starting (an expensive operation)
+// across the network, which can impact finalization.
+func (c *Controller) preStartDelay() time.Duration {
+	delay := computePreprocessingDelay(c.config.BaseStartDelay, c.dkg.Size())
+	return delay
+}
+
+// preHandleBroadcastDelay returns a duration to delay prior to handling EACH
+// broadcast message. This delay is used only during phase 1 of the DKG.
+// This prevents synchronization of processing verification vectors (an
+// expensive operation) across the network, which can impact finalization.
+func (c *Controller) preHandleBroadcastDelay() time.Duration {
+	delay := computePreprocessingDelay(c.config.BaseHandleBroadcastDelay, c.dkg.Size())
+	return delay
+}
+
+// computePreprocessingDelay computes a random delay to introduce before an
+// expensive operation.
+//
+// The maximum delay is m=b*n^2 where:
+// * b is a configurable base delay
+// * n is the size of the DKG committee
+func computePreprocessingDelay(baseDelay time.Duration, dkgSize int) time.Duration {
+	// sanity checks
+	if baseDelay < 0 {
+		baseDelay = 0
+	}
+	if dkgSize < 0 {
+		dkgSize = 0
+	}
+
+	// m=b*n^2
```

Comment:
Don't quite understand this math. Could you explain ?

Reply:
`maximum_delay = base_delay * number_of_dkg_nodes ^ 2`

Then we select the actual delay from `[0, maximum_delay]`

Reply:
What this PR is trying to do is spread out the CPU-intensive operations required in the DKG for each, not to spread out the actual broadcast messages. The broadcast messages all get sent at more or less the same time. Each node will receive `n-1` broadcast messages, then will stagger the processing of them to minimize the impact on the rest of the system.

The reason we use `n^2` is because the cost for a node to process an individual broadcast messages scales exponentially with the number of nodes and we want the aggregate amount of jitter introduced to be proportional to the aggregate amount of time spent processing broadcast messages.